### PR TITLE
[#366] not all email templates are saved in database

### PR DESCRIPTION
**Pull request by Pavliuk13**
https://user-images.githubusercontent.com/59260347/190377818-4ecff7a7-594c-4804-b593-dcf1516217d2.mp4

### CODE REVIEW

File: frontend/src/app/modules/availability/new-slot/notification-emails/notification-emails.component.ts [hunk 1]
```diff
@@ -13,29 +13,20 @@ import { ISaveConfirmationEmailDetails } from '@core/models/save-availability-sl
...
+        this.settings = this.slot?.emailTemplateSettings ?? getDefaultNotificationTemplates();
     }
 
-    private _defaultSettings: ISaveConfirmationEmailDetails = {
-        allowToSend: false,
-        type: TemplateType.Confirmation,
-        emailBody: getNotificationTemplate(TemplateType.Confirmation)?.body,
-        emailSubject: getNotificationTemplate(TemplateType.Confirmation)?.label,
-    };
+    public templateIndex: number = 0;
```
Comment (Ymkavi) [inline frontend/src/app/modules/availability/new-slot/notification-emails/notification-emails.component.ts:19]: Do we really need to use a numeric index, why not use a template type enum?

File: frontend/src/app/modules/availability/new-slot/notification-emails/notification-emails.component.ts [hunk 2]
```diff
@@ -58,13 +49,9 @@ export class NotificationEmailsComponent {
...
         this.activeTab = path;
-        this.currentSetting = this.slot?.emailTemplateSettings?.find((obj) => obj.type === this.settings.type);
-        this.settings = this.currentSetting ?? this._defaultSettings;
     }
 
     private changeTemplateType(type: TemplateType) {
-        this.settings.type = type;
-        this.settings.emailSubject = getNotificationTemplate(type)?.label;
-        this.settings.emailBody = getNotificationTemplate(type)?.body;
+        this.templateIndex = this.settings.indexOf(this.settings.find((el) => el.type === type)!);
```
Comment (Ymkavi) [inline frontend/src/app/modules/availability/new-slot/notification-emails/notification-emails.component.ts]: Try to reduce the use of "!" operator

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 3]
```diff
@@ -328,15 +325,20 @@ public async Task<bool> ValidateLinkAsync(long? slotId, string slotLink)
...
+                {
+                    dest.AvailabilitySlot = slot;
+                }))).ToList();
+            
+            await _context.EmailTemplates.AddRangeAsync(emailTemplates);
+            
+            foreach (var el in emailTemplates)
+            {
+                slot.EmailTemplates.Add(el);
+            }
```
Comment (Ymkavi) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]:
```suggestion
            slot.EmailTemplates = emailTemplates;
```

**Review by Ymkavi — CHANGES_REQUESTED** [review]
Now it is not possible to edit templates of already created slots

**Review by Ymkavi — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 4]
```diff
@@ -328,15 +334,17 @@ public async Task<bool> ValidateLinkAsync(long? slotId, string slotLink)
...
-                    }));
-            await _context.EmailTemplates.AddAsync(emailTemplate);
-            slot.EmailTemplates.Add(emailTemplate);
+            var emailTemplates = settingsDto.Select(s => _mapper.Map<EmailTemplate>(s,
+                opts => opts.AfterMap((_, dest) =>
+                {
+                    dest.AvailabilitySlot = slot;
+                }))).ToList();
+            
+            await _context.EmailTemplates.AddRangeAsync(emailTemplates);
```
Comment (apofis96) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs:345]: Doesn't it work without toList?

**Review by apofis96 — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs (hunk 4 above)
Comment (Pavliuk13) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs:345]: it works without list but my IDE gave me warning 'enumeration operation may not execute'
and '.ToList()' fixed it

**Review by apofis96 — APPROVED** [review]

**Pull request merged**